### PR TITLE
Pt no ptrs

### PR DESCRIPTION
**Pull request by MadWasp79**

### CODE REVIEW

File: native/src/transportRouteResultSegment.cpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	for (int32_t k = start; k < end; k++) {
+		const auto &stop = route->forwardStops[k];
+		const auto &nextStop = route->forwardStops[k + 1];
+		d += getDistance(stop->lat, stop->lon, nextStop->lat, nextStop->lon);
+	}
+	return d;
+}
+
+vector<Way> TransportRouteResultSegment::getGeometry() {
+	vector<Way> list;
```
Comment (vshcherb) [inline native/src/transportRouteResultSegment.cpp:38]: every time is recalculated no caching?

File: native/src/transportRoutingObjects.h [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+		return id == n.id && lat == n.lat && lon == n.lon;
+	}
+
+	friend std::size_t hash_value(Node const &n);
+};
+
+struct Way {
+	int64_t id;
+	vector<Node> nodes;
+	vector<int64_t> nodeIds;
```
Comment (vshcherb) [inline native/src/transportRoutingObjects.h:93]: nodeIds  field not used at all?

File: native/src/transportRoutingObjects.h [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+struct TransportRoute : public MapObject {
+	vector<SHARED_PTR<TransportStop>> forwardStops;
+	string ref;
+	string routeOperator;
+	string type;
+	uint32_t dist;
+	string color;
+	vector<Way> forwardWays;
+	SHARED_PTR<TransportSchedule> schedule;
```
Comment (vshcherb) [inline native/src/transportRoutingObjects.h:144]: why shared_ptr is needed at all? just object should be fine

File: native/src/transportRoutingObjects.h [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+	TransportRoute();
+
+	SHARED_PTR<TransportSchedule> getOrCreateSchedule();
```
Comment (vshcherb) [inline native/src/transportRoutingObjects.h:148]: should be TransportSchedule&

File: native/src/transportRouteResult.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+#define _OSMAND_TRANSPORT_ROUTE_RESULT_H
+#include "CommonCollections.h"
+#include "commonOsmAndCore.h"
+
+struct TransportRouteResultSegment;
+struct TransportRoutingContext;
+struct TransportRoutingConfiguration;
+
+struct TransportRouteResult {
+	vector<SHARED_PTR<TransportRouteResultSegment>> segments;
```
Comment (vshcherb) [inline native/src/transportRouteResult.h:11]: Shared_ptr ?

File: native/src/transportRoutePlanner.cpp [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+		}
+		if (!include) {
+			lst.push_back(route);
+			// System.out.println(route.toString());
+		}
+	}
+	return lst;
+}
+
+vector<SHARED_PTR<TransportRouteResult>>
```
Comment (vshcherb) [inline native/src/transportRoutePlanner.cpp:124]: shared_ptr<TransportRouteResult> ?

File: native/src/transportRouteResult.h (hunk 5 above)
Comment (MadWasp79) [reply native/src/transportRouteResult.h]: fixed

File: native/src/transportRoutePlanner.cpp (hunk 6 above)
Comment (MadWasp79) [reply native/src/transportRoutePlanner.cpp]: fixed

File: native/src/transportRoutingObjects.h (hunk 2 above)
Comment (MadWasp79) [reply native/src/transportRoutingObjects.h:93]: removed

File: native/src/transportRoutingObjects.h (hunk 4 above)
Comment (MadWasp79) [reply native/src/transportRoutingObjects.h:144]: done

File: native/src/transportRoutingObjects.h (hunk 3 above)
Comment (MadWasp79) [reply native/src/transportRoutingObjects.h:140]: done

File: native/src/transportRouteResultSegment.cpp (hunk 1 above)
Comment (MadWasp79) [reply native/src/transportRouteResultSegment.cpp]: fixed

**Pull request merged**